### PR TITLE
Add general email event for caseworkers

### PR DESCRIPTION
**Pull request by nitinprabhu**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-781

### Change description ###

- Added general order email event.
- Includes unit, integration and functional tests.
- Notify templates are added and template ids are configured in application config.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/notification/GeneralEmailNotification.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                emailTo = caseData.getApplicant1().getSolicitor().getEmail();
+                templateId = GENERAL_EMAIL_PETITIONER_SOLICITOR;
+                templateVars.put("solicitor name", caseData.getApplicant1().getSolicitor().getName());
+            } else {
+                log.info("Sending General Email Notification to petitioner for case id: {}", caseId);
+                emailTo = caseData.getApplicant1().getEmail();
+                templateId = GENERAL_EMAIL_PETITIONER;
+            }
+        } else if (RESPONDENT.equals(caseData.getGeneralEmail().getGeneralEmailParties())) {
+            if (caseData.getApplicant2().isRepresented()) {
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/service/notification/GeneralEmailNotification.java:55]: Could applicant2 ever be null and generalEmailParties be set to RESPONDENT?

Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/notification/GeneralEmailNotification.java:55]: I think we will always have applicant 1 and applicant 2 so should not be an issue. When you create application I think we need both petitioner and solicitor details.
@linusnorton could you please confirm just to be on safer side :)

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralEmail.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+@Slf4j
+public class CaseworkerGeneralEmail implements CCDConfig<CaseData, State, UserRole> {
+    public static final String CASEWORKER_CREATE_GENERAL_EMAIL = "caseworker-create-general-email";
+
+    @Autowired
+    private GeneralEmailNotification generalEmailNotification;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerGeneralEmail.java:36]: I'm confused, where is the closing )

File: src/main/java/uk/gov/hmcts/divorce/caseworker/service/notification/GeneralEmailNotification.java (hunk 1 above)
Comment (linusnorton) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/service/notification/GeneralEmailNotification.java:55]: Correct, we will always have an applicant2 after the application has been submitted and the caseworker can't see the case while it's in draft

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralEmail.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class GeneralEmail {
+
+    @CCD(
+        label = "Address to",
+        access = {CaseworkerAccessBetaOnlyAccess.class},
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralEmail.java:22]: You could either set the access on the property in CaseData or not set it at all and rely on the permissions being picked up from the event permissions

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralOrderDivorceParties.java [hunk 4]
```diff
@@ -12,10 +12,7 @@
     PETITIONER("Petitioner"),
 
     @JsonProperty("respondent")
-    RESPONDENT("Respondent"),
-
-    @JsonProperty("coRespondent")
-    CO_RESPONDENT("Co-respondent");
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralOrderDivorceParties.java:18]: Linus Norton endorses this change

**Review by linusnorton — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralEmail.java (hunk 3 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/divorce/divorcecase/model/GeneralEmail.java:22]: didn't realised that will remove it from here and will rely on event permissions.

**Pull request merged**